### PR TITLE
Adding tape and end to end test

### PR DESCRIPTION
**Pull request by ynachiket**
npm install 
npm test

(Assumes you have docker and that you are on mac. This is just to get started)

What does npm test do:

Uses @robert-chiniquy 's depends-on to spin up docker containers for postgres and noit+stratcon
Test1: checks if @lekhajee 's noit_client can connect to the noit process
Test2: checks if the journals are accessible over http to consume 
Cleans up docker containers after we are done

Next up:

Use this test setup to build a nodejs stratcon client to consume journals and in a way to enable running tests against the data.

Update:
Includes Stratcon ingestor MVP
Includes basic ingestion tests

Want to open this up to discussion so that I can get some help on future designs.

### CODE REVIEW

File: tests/test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+var ready = require('depends-on')(['postgres', 'noit_stratcon']);
+var cleanup = require('depends-on')(['kill_postgres', 'kill_noit_stratcon', 'remove_postgres', 'remove_noit']);
+var test = require('tape');
+var http = require('http');
+var client = require('noit_client').NoitClient;
+
+test("dependencies", ready);
+
+test('test noit', function(t) {
+	var noit_client = new client('192.168.59.103', '8888', null, null, null);
```
Comment (rphillips) [inline tests/test.js]: nit, leading spaces

Comment (rphillips) [inline tests/test.js]: should be 127.0.0.1

Comment (rphillips) [inline tests/test.js]: could leave off the nulls perhaps

File: tests/test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	  if (response) {
+	    console.log("getNoitVersion Response: ", response)
+	  }
+	  t.ok(response);
+	  t.end();
+  })
+});
+
+test('test journal', function(t) {
+	http.get("http://192.168.59.103/handoff/journals", function(res) {
```
Comment (rphillips) [inline tests/test.js]: 127.0.0.1 w/port

File: tests/test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      console.log("getNoitVersion Response: ", response)
+    }
+    t.ok(response);
+    t.end();
+  })
+});
+
+test('test journal', function(t) {
+  http.get("http://" + ip + "/handoff/journals", function(res) {
+    console.log("Got response: " + res.statusCode);
```
Comment (lekhajee) [inline tests/test.js]: why `console.log` and not `t.ok` on the `statusCode`

File: package.json [hunk 4]
```diff
@@ -17,7 +17,9 @@
   },
   "homepage": "https://github.com/virgo-agent-toolkit/agent-poller-noit-parity",
   "dependencies": {
-    "depends-on": "0.0.2",
-    "tape": "^2.13.0"
+    "depends-on": "^1.1.0",
+    "longjohn": "~0.2.2",
+    "tape": "^2.13.0",
```
Comment (lekhajee) [inline package.json:22]: these should be under `devDependencies`

File: tests/test.js (hunk 3 above)
Comment (ynachiket) [inline tests/test.js]: this is just a proof of concept.. these are not actual tests. 
but noted. changed.

File: package.json (hunk 4 above)
Comment (ynachiket) [inline package.json:22]: this being a test framework, do we still want to have separate dependencies?

Comment (simonvetter) [inline package.json:22]: good point :)

**Comment by ynachiket** [comment]
Closing this out as all this logic was moved to the noit_client
https://github.com/virgo-agent-toolkit/noit_client/commit/37a16eab5daf44807a7499457a71be88dabce498

**Pull request closed without merging**